### ISSUE TITLE
Polygon-to-Polygon euclidean_distance hangs forever

### ISSUE THREAD

**Issue by leops** [georust/geo#555]
I came across two polygons that do not seem to exhibit any "weird" characteristics (both are convex, have only a few points and do not overlap), but for some reason they hang the `euclidean_distance` function indefinitely.

# Reproduction

```rust
use geo::{polygon, prelude::*};

fn main() {
    let a = polygon!(
        exterior: [
            (x: -1.5350399, y: -0.18596762),
            (x: -0.33726108, y: 0.2206358),
            (x: -0.13836576, y: 0.42173427),
            (x: -0.33946428, y: 0.6206298),
            (x: -1.1405537, y: 0.8162205),
            (x: -1.3405507, y: 0.81511897),
            (x: -1.5372429, y: 0.21402647),
        ],
        interiors: []
    );

    let b = polygon!(
        exterior: [
            (x: 0.6649223, y: -0.17384565),
            (x: 0.8649193, y: -0.1727441),
            (x: 1.0616114, y: 0.4283484),
            (x: 0.8616146, y: 0.4272468),
            (x: 0.66271913, y: 0.22614834),
        ],
        interiors: []
    );

    a.euclidean_distance(&b);
}
```

Perhaps interestingly, calling `b.euclidean_distance(&a)` works correctly.

**Comment by urschrei** [comment]
Hmm. This is almost certainly an issue with the rotating calipers code.

**Comment by msalib** [comment]
Here's a smaller test case consisting of two disjoint rectangles that have been converted into polygons. `euclidean_distance` hangs because `min_poly_dist` hangs.

```rust
        let a = Rect::new((0., 0.), (1., 1.)).to_polygon();
        let b = Rect::new((2., 2.), (3., 3.)).to_polygon();
        min_poly_dist(&a, &b);
```

I have no idea what the fix is, but since this is supposed to be a linear algorithm, perhaps we can find a bound on the expected number of iterations and panic when we exceed that?

**Comment by urschrei** [comment]
That's a good plan, but I'd be equally happy to just disable the fast path until I get around to investigating / debugging, which will probably be at least a couple of months away.

**Comment by urschrei** [comment]
Fixed in #731. Apologies for the delay @leops.
